onnx/tensor.h

```diff
@@ -0,0 +1,141 @@
+#pragma once
```

Comment:
Do we want to use tensor structure defined in https://github.com/dmlc/dlpack/blob/master/include/dlpack/dlpack.h as proposed before please? @dzhulgakov

Reply:
That struct is more for a interchange of data between APIs. Directly using it is annoying - you'd still need to allocate memory, do i/o, etc.

Reply:
Let's keep track this topic here: https://github.com/onnx/onnx/issues/438.

If we decide to add support for dlpack, it's better to have another PR.